src/native-common/TextInput.tsx

```diff
@@ -192,27 +198,20 @@ export class TextInput extends React.Component<Types.TextInputProps, TextInputSt
     }
 
     selectAll() {
-        this._selectionStart = 0;
-        this._selectionEnd = this.state.inputValue.length;
+        this._selectionToSet = { start: 0, end: this.state.inputValue.length };
         this.forceUpdate();
     }
 
     selectRange(start: number, end: number) {
         const constrainedStart = Math.min(start, this.state.inputValue.length);
         const constrainedEnd = Math.min(end, this.state.inputValue.length);
 
-        this._selectionStart = constrainedStart;
-        this._selectionEnd = constrainedEnd;
+        this._selectionToSet = { start: constrainedStart, end: constrainedEnd };
         this.forceUpdate();
     }
 
     getSelectionRange(): { start: number, end: number } {
-        let range = {
-            start: this._selectionStart,
-            end: this._selectionEnd
-        };
-
-        return range;
+        return this._selection;
```

Comment:
I don't see any place where this._selection is updated. It appears that it will always return { start: 0, end: 0 } regardless of the current selection.

Reply:
It is updated in _onSelectionChange, line 161

Reply:
Ah yes, I missed that. Looks good.